### PR TITLE
Added an axis param to support y-axis events

### PR DESCRIPTION
**Pull request by jrowen**
This update enables y-axis events, which can also be used to add a horizontal line to a chart.
```
dygraph(presidents, main = "Presidential Approval") %>%
    dyAxis("y", valueRange = c(0, 100)) %>%
     dyEvent("1950-6-30", "Korea", labelLoc = "bottom") %>%
     dyEvent("1965-2-09", "Vietnam", labelLoc = "bottom") %>% 
     dyEvent(max(presidents, na.rm = TRUE), "Max", axis = "y", 
             strokePattern = "solid", color = "blue")
```

### CODE REVIEW

**Comment by jrowen** [comment]
A y-axis line would probably be better categorized as a level than an event, so instead of adding the axis param to dyEvent, it may be more readable to create a dyLevel function and set the axis param within the function logic.  Any thoughts?

**Comment by jjallaire** [comment]
Yes I agree that this is better introduced as a new dyLevel function.

**Comment by jjallaire** [comment]
In doing this I'd keep the JavaScript side pretty much as you have it (one code path that has conditional branches for x and y) and just introduce a new wrapper R function (e.g. `dyLevel`)

File: R/event.R [hunk 1]
```diff
@@ -36,11 +36,12 @@ dyEvent <- function(dygraph,
   
   # create event
   event <- list()
-  event$date <- asISO8601Time(date)
-  event$label <- label
+  event$pos = asISO8601Time(date)
+  if (!missing(label) && !is.null(label)) event$label <- label
```
Comment (jjallaire) [inline R/event.R]: Why this treatment rather than just having label = NULL as the default argument value?

Comment (jjallaire) [inline R/event.R]: My thinking is that with `label = NULL` the fact that the parameter is optional is self-documenting from the signature (this is also the idiom we use of optional parameters throughout the rest of the package).

**Pull request merged**